vm/src/protocol/number.rs

```diff
@@ -441,71 +460,101 @@ impl PyNumber<'_> {
     pub fn int(self, vm: &VirtualMachine) -> Option<PyResult<PyIntRef>> {
         self.class().slots.as_number.int.load().map(|f| {
             let ret = f(self, vm)?;
-            let value = if !ret.class().is(PyInt::class(&vm.ctx)) {
-                warnings::warn(
-                    vm.ctx.exceptions.deprecation_warning,
-                    format!(
-                        "__int__ returned non-int (type {}).  \
-                The ability to return an instance of a strict subclass of int \
-                is deprecated, and may be removed in a future version of Python.",
-                        ret.class()
-                    ),
-                    1,
-                    vm,
-                )?;
-                vm.ctx.new_bigint(ret.as_bigint())
+
+            if let Some(ret) = ret.downcast_ref_if_exact::<PyInt>(vm) {
+                Ok(vm.ctx.new_pyref(ret.as_bigint().to_owned()))
```

Comment:
```suggestion
                return Ok(vm.ctx.new_pyref(ret.as_bigint().to_owned()));
```
I also like to use `return` here to keep the later code dedented one more depth